packages/babel-helper-function-name/src/index.ts

```diff
@@ -223,6 +223,8 @@ export default function (
     name = id.name;
   }
 
+  if (isFunction(node) && /[\u{10000}-\u{10ffff}]/u.test(name)) return;
```

Comment:
Just curious: won't this get transpiled into an unnecessarily complex regex without unicode flag?
I mean, `/[\uD800-\uDFFF]/.test(name)` would work as well, right?

Reply:
@nicolo-ribaudo suggested me to use this #14047 in this issue.

Reply:
We transpile our code targeting Node.js 6, so we don't transpile the `u` flag. However yes, `/[\uD800-\uDFFF]/.test(name)` would work too.

Reply:
BTW `/[\uD800-\uDFFF]/` are not equivalent to `/[\u{10000}-\u{10ffff}]/u` because lone surrogate can be a valid string literal. 🤦
```js
var o = {
  "\uD800"() {}
}
```

Reply:
Ok so `/[\uD800-\uDFFF]/` is the correct one in this case.

Reply:
Okay, I will change it to `/[\uD800-\uDFFF]/`.